src/main/java/com/metamx/common/StringUtils.java

```diff
@@ -66,4 +67,21 @@ public static String fromUtf8(final ByteBuffer buffer)
       throw Throwables.propagate(e);
     }
   }
+
+  public static String safeFormat(String message, Object... formatArgs)
+  {
+    if(formatArgs == null || formatArgs.length == 0) {
+      return message;
+    }
+    try {
+      return String.format(message, formatArgs);
```

Comment:
As you noted below, this is a direct refactor out of Logger

Suggest not changing this method in this PR since it is a direct move.